public/app/features/datasources/hooks.ts

```diff
@@ -28,7 +28,12 @@ export function useRecentlyUsedDataSources(): [string[], (ds: DataSourceInstance
         );
         setStorage([...value, ds.uid]);
       } else {
-        setStorage([...value, ds.uid].slice(1, 6));
+        const newArray = [...value, ds.uid];
+        if (newArray.length > 5) {
+          setStorage(newArray.slice(1, 6));
+        } else {
+          setStorage(newArray);
+        }
```

Comment:
Should we add a test case for the logic that was fixed? (If we have a single item in the list then doesn't remove it and return with an empty list.)

Reply:
yep, it's here already: https://github.com/grafana/grafana/pull/108971/files#diff-bf7a33c0877a3e9069077d5adcb1fdab2e53f05ca65bcfcd74b99bec07c57c5eR44

Reply:
Hmm, is there? What I meant is testing if adding a datasource works in case the array was empty previously (I guess that was the issue this PR fixed - the `.slice(1, 6)` cutting the first item -, right?). I think `describe('maximum items limit', () => {` only tests if you can shift in a datasource if there were already five uids in the list?

Not too important though, just to see if I misunderstood something.

Reply:
yes, maybe the link pointed you to a different place, what I meant is the test "should add a new data source to an empty list", which does that, starting from a empty list, it adds one datasource (and verifies it's stored).

Reply:
Aaa, superb, it was me then, thanks 🙏